snuba/subscriptions/scheduler.py

```diff
@@ -321,6 +329,36 @@ def __reset_builder(self) -> None:
             # We are transitioning between jittered and immediate mode. We must use the delegate builder.
             self.__builder = self.__delegate_builder
 
+    def __get_filtered_subscriptions(self) -> List[Subscription]:
+        if self.__slice_id is not None:
```

Comment:
It would be better if we move the slice_id checks in __get_subscriptions and call __get_filtered_subscriptions only in the case of slice_id being enabled.